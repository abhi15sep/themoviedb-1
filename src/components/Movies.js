import React from 'react';
import PropTypes from 'prop-types'
import Movie from './Movie';
import { Grid, Row } from 'react-bootstrap';

function Movies(props) {
  const movies = props.data.map((movie) => {
    return <Movie key={movie.id} data={movie} />;
  });

  return (
    <Grid>
      <Row>
        {movies}
      </Row>
    </Grid>
  );
}

Movies.propTypes = {
  data: PropTypes.arrayOf(PropTypes.object).isRequired,
}

export default Movies;
