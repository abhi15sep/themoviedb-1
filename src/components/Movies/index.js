import React from 'react';
import PropTypes from 'prop-types';
import { Container, Row } from 'react-bootstrap';
import Movie from '../Movie';
import Types from '../../utils/types';
import './styles.css';

/**
 * Component for display the movie list
 *
 * @param {object} props - The props of the component
 * @param {object} props.movies - The movie list
 * @returns {Movies} - The react component
 */
export default function Movies({ movies }) {
  return (
    <Container className="Movies-main">
      <Row>
        {movies.map((movie) => <Movie key={movie.id} movie={movie} />)}
      </Row>
    </Container>
  );
}

Movies.propTypes = {
  movies: PropTypes.arrayOf(Types.movie.isRequired).isRequired,
};
