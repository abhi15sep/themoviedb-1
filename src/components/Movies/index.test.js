import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router';
import Movies from './index';

const movies = [
  {
    id: 1,
    vote_average: 5,
    title: 'example 1',
    overview: 'Test',
    poster_path: '/path/to/image',
    release_date: '2017-04-28',
  },
  {
    id: 2,
    vote_average: 3,
    title: 'example 2',
    overview: 'Test',
    poster_path: '/path/to/image',
    release_date: '2017-04-28',
  },
  {
    id: 3,
    vote_average: 9,
    title: 'example 3',
    overview: 'Test',
    poster_path: '/path/to/image',
    release_date: '2017-04-28',
  },
];

test('renders correctly', () => {
  const component = renderer.create(<Router><Movies movies={movies} /></Router>);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
