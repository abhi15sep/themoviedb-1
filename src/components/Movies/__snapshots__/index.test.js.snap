// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Movies-main container"
>
  <div
    className="row"
  >
    <div
      className="col-lg-6 col-sm-12"
    >
      <div
        className="Movie-card"
      >
        <div
          className="row"
        >
          <div
            className="col-md-4 col-sm-12"
          >
            <a
              href="/details/1"
              onClick={[Function]}
            >
              <img
                alt="example 1"
                src="http://image.tmdb.org/t/p/w185/path/to/image"
                title="example 1"
              />
            </a>
          </div>
          <div
            className="col-md-8 col-sm-12"
          >
            <div
              className="Movie-body"
            >
              <div
                className="row"
              >
                <div
                  className="col-1"
                >
                  <span
                    className="badge badge-danger"
                  >
                    5
                  </span>
                </div>
                <div
                  className="col-10"
                >
                  <a
                    href="/details/1"
                    onClick={[Function]}
                  >
                    <span
                      className="Movie-title"
                    >
                      example 1
                    </span>
                  </a>
                  <br />
                  <small>
                    April 28, 2017
                  </small>
                </div>
              </div>
              <p
                className="Movie-overview"
              >
                Test
                ...
              </p>
              <div
                className="Movie-details"
              >
                <a
                  className="Movie-details-link"
                  href="/details/1"
                  onClick={[Function]}
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    strokeWidth="0"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
                    />
                  </svg>
                   
                  <small>
                    More info
                  </small>
                </a>
                <div
                  className="Recommendation-main"
                >
                  <small>
                    <svg
                      className="Recommendation-icon"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      strokeWidth="0"
                      style={
                        Object {
                          "color": undefined,
                        }
                      }
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M0 56v240c0 13.255 10.745 24 24 24h80c13.255 0 24-10.745 24-24V56c0-13.255-10.745-24-24-24H24C10.745 32 0 42.745 0 56zm40 200c0-13.255 10.745-24 24-24s24 10.745 24 24-10.745 24-24 24-24-10.745-24-24zm272 256c-20.183 0-29.485-39.293-33.931-57.795-5.206-21.666-10.589-44.07-25.393-58.902-32.469-32.524-49.503-73.967-89.117-113.111a11.98 11.98 0 0 1-3.558-8.521V59.901c0-6.541 5.243-11.878 11.783-11.998 15.831-.29 36.694-9.079 52.651-16.178C256.189 17.598 295.709.017 343.995 0h2.844c42.777 0 93.363.413 113.774 29.737 8.392 12.057 10.446 27.034 6.148 44.632 16.312 17.053 25.063 48.863 16.382 74.757 17.544 23.432 19.143 56.132 9.308 79.469l.11.11c11.893 11.949 19.523 31.259 19.439 49.197-.156 30.352-26.157 58.098-59.553 58.098H350.723C358.03 364.34 384 388.132 384 430.548 384 504 336 512 312 512z"
                      />
                    </svg>
                    Bad choice
                  </small>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-lg-6 col-sm-12"
    >
      <div
        className="Movie-card"
      >
        <div
          className="row"
        >
          <div
            className="col-md-4 col-sm-12"
          >
            <a
              href="/details/2"
              onClick={[Function]}
            >
              <img
                alt="example 2"
                src="http://image.tmdb.org/t/p/w185/path/to/image"
                title="example 2"
              />
            </a>
          </div>
          <div
            className="col-md-8 col-sm-12"
          >
            <div
              className="Movie-body"
            >
              <div
                className="row"
              >
                <div
                  className="col-1"
                >
                  <span
                    className="badge badge-danger"
                  >
                    3
                  </span>
                </div>
                <div
                  className="col-10"
                >
                  <a
                    href="/details/2"
                    onClick={[Function]}
                  >
                    <span
                      className="Movie-title"
                    >
                      example 2
                    </span>
                  </a>
                  <br />
                  <small>
                    April 28, 2017
                  </small>
                </div>
              </div>
              <p
                className="Movie-overview"
              >
                Test
                ...
              </p>
              <div
                className="Movie-details"
              >
                <a
                  className="Movie-details-link"
                  href="/details/2"
                  onClick={[Function]}
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    strokeWidth="0"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
                    />
                  </svg>
                   
                  <small>
                    More info
                  </small>
                </a>
                <div
                  className="Recommendation-main"
                >
                  <small>
                    <svg
                      className="Recommendation-icon"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      strokeWidth="0"
                      style={
                        Object {
                          "color": undefined,
                        }
                      }
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M0 56v240c0 13.255 10.745 24 24 24h80c13.255 0 24-10.745 24-24V56c0-13.255-10.745-24-24-24H24C10.745 32 0 42.745 0 56zm40 200c0-13.255 10.745-24 24-24s24 10.745 24 24-10.745 24-24 24-24-10.745-24-24zm272 256c-20.183 0-29.485-39.293-33.931-57.795-5.206-21.666-10.589-44.07-25.393-58.902-32.469-32.524-49.503-73.967-89.117-113.111a11.98 11.98 0 0 1-3.558-8.521V59.901c0-6.541 5.243-11.878 11.783-11.998 15.831-.29 36.694-9.079 52.651-16.178C256.189 17.598 295.709.017 343.995 0h2.844c42.777 0 93.363.413 113.774 29.737 8.392 12.057 10.446 27.034 6.148 44.632 16.312 17.053 25.063 48.863 16.382 74.757 17.544 23.432 19.143 56.132 9.308 79.469l.11.11c11.893 11.949 19.523 31.259 19.439 49.197-.156 30.352-26.157 58.098-59.553 58.098H350.723C358.03 364.34 384 388.132 384 430.548 384 504 336 512 312 512z"
                      />
                    </svg>
                    Bad choice
                  </small>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-lg-6 col-sm-12"
    >
      <div
        className="Movie-card"
      >
        <div
          className="row"
        >
          <div
            className="col-md-4 col-sm-12"
          >
            <a
              href="/details/3"
              onClick={[Function]}
            >
              <img
                alt="example 3"
                src="http://image.tmdb.org/t/p/w185/path/to/image"
                title="example 3"
              />
            </a>
          </div>
          <div
            className="col-md-8 col-sm-12"
          >
            <div
              className="Movie-body"
            >
              <div
                className="row"
              >
                <div
                  className="col-1"
                >
                  <span
                    className="badge badge-success"
                  >
                    9
                  </span>
                </div>
                <div
                  className="col-10"
                >
                  <a
                    href="/details/3"
                    onClick={[Function]}
                  >
                    <span
                      className="Movie-title"
                    >
                      example 3
                    </span>
                  </a>
                  <br />
                  <small>
                    April 28, 2017
                  </small>
                </div>
              </div>
              <p
                className="Movie-overview"
              >
                Test
                ...
              </p>
              <div
                className="Movie-details"
              >
                <a
                  className="Movie-details-link"
                  href="/details/3"
                  onClick={[Function]}
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    strokeWidth="0"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
                    />
                  </svg>
                   
                  <small>
                    More info
                  </small>
                </a>
                <div
                  className="Recommendation-main"
                >
                  <small>
                    <svg
                      className="Recommendation-icon"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      strokeWidth="0"
                      style={
                        Object {
                          "color": undefined,
                        }
                      }
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M104 224H24c-13.255 0-24 10.745-24 24v240c0 13.255 10.745 24 24 24h80c13.255 0 24-10.745 24-24V248c0-13.255-10.745-24-24-24zM64 472c-13.255 0-24-10.745-24-24s10.745-24 24-24 24 10.745 24 24-10.745 24-24 24zM384 81.452c0 42.416-25.97 66.208-33.277 94.548h101.723c33.397 0 59.397 27.746 59.553 58.098.084 17.938-7.546 37.249-19.439 49.197l-.11.11c9.836 23.337 8.237 56.037-9.308 79.469 8.681 25.895-.069 57.704-16.382 74.757 4.298 17.598 2.244 32.575-6.148 44.632C440.202 511.587 389.616 512 346.839 512l-2.845-.001c-48.287-.017-87.806-17.598-119.56-31.725-15.957-7.099-36.821-15.887-52.651-16.178-6.54-.12-11.783-5.457-11.783-11.998v-213.77c0-3.2 1.282-6.271 3.558-8.521 39.614-39.144 56.648-80.587 89.117-113.111 14.804-14.832 20.188-37.236 25.393-58.902C282.515 39.293 291.817 0 312 0c24 0 72 8 72 81.452z"
                      />
                    </svg>
                    Editor choice
                  </small>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
